auth/auth.go

```diff
@@ -1,21 +1,50 @@
 package auth
 
 import (
+	"io/ioutil"
+	"net/http"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/golang/glog"
 	"github.com/rancherio/host-api/app/common"
 	"github.com/rancherio/host-api/config"
-	jwt "github.com/dgrijalva/jwt-go"
-	"net/http"
 )
 
+var fileBytes []byte
+
 func Auth(rw http.ResponseWriter, req *http.Request) bool {
 	if !config.Config.Auth {
+		glog.Infoln("Host_API auth is disabled")
 		return true
 	}
+	getToken := req.URL.Query().Get("token")
 
-	token, err := jwt.ParseFromRequest(req, func(token *jwt.Token) (interface{}, error) {
-		return config.Config.Key, nil
+	if len(getToken) == 0 {
+		return false
+	}
+	glog.Infoln("Enabling Authencation for Host_API")
+
+	if len(fileBytes) == 0 {
```

Comment:
The logic of caching should be done on startup of host-api.  The logical place would be in the config.Parse().  Besides just reading the bytes you should parse the bytes to a `*rsa.PublicKey` from `crypto/rsa`.  The current approach requires you to parse the rsa on each request.